gms/persistent_feature_enabler.hh

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <seastar/core/sstring.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/semaphore.hh>
+#include <set>
+#include "seastarx.hh"
+
+namespace db {
+class system_keyspace;
+}
+
+namespace gms {
+
+class feature_service;
+
+// Responsible for enabling features in a persistent fashion.
+// Note that this is not a service! All methods should be called
```

Comment:
I would say it is a service, but not a `sharded` one.

Reply:
Right. The service will most likely be completely removed in v2 though, see https://github.com/scylladb/scylladb/pull/14722#discussion_r1270345596.